packages/core/src/shared/remoteSession.ts

```diff
@@ -175,6 +181,61 @@ export async function handleMissingTool(tools: Err<MissingTool[]>) {
     )
 }
 
+function getFormattedSsmActions() {
+    const formattedActions = minimumSsmActions.map((action) => `"${action}",\n`).reduce((l, r) => l + r)
+
+    return formattedActions.slice(0, formattedActions.length - 2)
+}
+
+async function addInlinePolicyWithDelay(client: IamClient, roleArn: string) {
+    const timeout = new Timeout(policyAttachDelay)
+    const message = `Adding Inline Policy to ${roleArn}`
+    await showMessageWithCancel(message, timeout)
```

Comment:
I experimented with this, but the behavior is slightly different. `showAfterMs` will delay the progress bar from popping up, whereas `delay` will extend the progress bar being up for some delay. The advantage of the second is that the moment the customer presses the button they receive feedback in a progress bar, rather than waiting for `policyAttachDelay` seconds to see any indication something is happening.

Reply:
Maybe that's worth adding to `showMessageWithCancel` as an optional feature?